iroh/src/magicsock/remote_map/remote_state.rs

```diff
@@ -671,6 +675,9 @@ impl RemoteStateActor {
                     ?local_candidates,
                     ?remote_candidates,
                 );
+                if self.last_holepunch.is_none() {
+                    self.metrics.remote_holepunch_attempts.inc();
+                }
```

Comment:
I think the counter you removed is much more helpful. All the logic in trigger_holepunching should not call this function if it is not needed. So every time we call this function increment an attempt counter should be enough.

Though it is really hard to match up attempts with successes. I'm not sure it's possible right now even. Can't believe I naively wrote that down in the issue 😅